app/models/application_record.rb

```diff
@@ -7,6 +7,18 @@ class ApplicationRecord < ActiveRecord::Base
   after_update { |record| log_analytics record, "updated" }
   after_destroy { |record| log_analytics record, "destroyed" }
 
+  # Condition for rollout of mass addition of validation rules.
+  # Memoizes for performance.
+  def mass_validation_enabled?
+    if AppConfig.table_exists?
+      if defined?(@enable_mass_validation)
```

Comment:
Your approach is caching that flag per object. That means if you create thousands of model objects, it will fetch the config table for each of them.

Although it is better than the previous implementation, I would suggest improving performance even further by using rails cache. I know that will force us to redeploy every time we change this value, but I don't think it is an issue given the nature of this flag.

My suggestion:
```
  def mass_validation_enabled?
    Rails.cache.fetch("ApplicationRecord.mass_validation_enabled?") do
      AppConfigHelper.get_app_config(AppConfig::ENABLE_MASS_VALIDATION)
    end
  end
```

Reply:
Thanks, good suggestion! Updated.